tslearn/metrics/cysax.py

```diff
@@ -0,0 +1,150 @@
+STUFF_cysax = "cysax"
+
+import numpy as np
+from numba import njit, prange
+from sklearn.linear_model import LinearRegression
+
+__author__ = "Romain Tavenard romain.tavenard[at]univ-rennes2.fr"
+
+DTYPE = float
+DTYPE_INT = int
+
+
+@njit(parallel=True)
+def inv_transform_paa(dataset_paa, original_size):
+    n_ts = dataset_paa.shape[0]
+    sz = dataset_paa.shape[1]
+    d = dataset_paa.shape[2]
+    i = 0
+    t = 0
+    di = 0
+    t0 = 0
+    seg_sz = original_size // sz
+    dataset_out = np.zeros((n_ts, original_size, d))
+
+    for i in prange(n_ts):
+        for t in range(sz):
+            t0 = t * seg_sz
+            for di in range(d):
+                dataset_out[i, t0 : t0 + seg_sz, di] = dataset_paa[i, t, di]
```

Comment:
can we replace the third for-loop with:
```
dataset_out[i, t0 : t0 + seg_sz, :] = dataset_paa[i, t, :]
```
?
Also, maybe we can use `np.repeat` to remove the second for-loop as well. (I am not sure though)

Reply:
Thank you. I have change the order of the loops, now I only have:
    for t in prange(sz):
        t0 = t * seg_sz
        dataset_out[:, t0: t0 + seg_sz, :] = dataset_paa[:, t, :]